zerver/lib/name_restrictions.py

```diff
@@ -1,7 +1,10 @@
 from disposable_email_domains import blacklist
+from django.conf import settings
 
 
 def is_reserved_subdomain(subdomain: str) -> bool:
+    if subdomain in settings.SOCIAL_AUTH_SUBDOMAIN:
+        return True
```

Comment:
I think you want `if string_id in [settings.SOCIAL_AUTH_SUBDOMAIN]:` -- it's not a list.  This is doing substring comparisons.  Or at least if that's intentional, we should have a comment noting it.

Reply:
Yeah this should be `==`, I'm pretty sure this was a mistake